apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/PartitionOptionsTests.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright 2023 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+using Google.Cloud.Spanner.V1;
+using System;
+using System.Threading;
+using Xunit;
+
+namespace Google.Cloud.Spanner.Data.Tests;
+
+public class PartitionOptionsTests
+{
+    [Fact]
+    public void CheckDefaultValues()
+    {
+        AssertDefaultOptions(PartitionOptions.Default);
+    }
+
+    [Fact]
+    public void Default_WithDataBoostEnabled_ShouldCreateNewInstanceWithUpdatedValue()
+    {
+        var defaultOptions = PartitionOptions.Default;
+        var newOptions = defaultOptions.WithDataBoostEnabled(true);
+
+        AssertDefaultOptions(defaultOptions);
+        Assert.Null(newOptions.MaxPartitions);
+        Assert.Null(newOptions.PartitionSizeBytes);
+        Assert.True(newOptions.DataBoostEnabled);
+    }
+
+    [Fact]
+    public void Default_WithMaxPartitions_ShouldCreateNewInstanceWithUpdatedValue()
+    {
+        var defaultOptions = PartitionOptions.Default;
+        var newOptions = defaultOptions.WithMaxPartitions(5);
+
+        AssertDefaultOptions(defaultOptions);
+        Assert.Equal(5, newOptions.MaxPartitions);
+        Assert.Null(newOptions.PartitionSizeBytes);
+        Assert.False(newOptions.DataBoostEnabled);
+    }
+
+    [Fact]
+    public void Default_WithPartitionSizeBytes_ShouldCreateNewInstanceWithUpdatedValue()
+    {
+        var defaultOptions = PartitionOptions.Default;
+        var newOptions = defaultOptions.WithPartitionSizeBytes(15);
+
+        AssertDefaultOptions(defaultOptions);
+        Assert.Null(newOptions.MaxPartitions);
+        Assert.Equal(15, newOptions.PartitionSizeBytes);
+        Assert.False(newOptions.DataBoostEnabled);
+    }
+
+    private void AssertDefaultOptions(PartitionOptions defaultOptions)
+    {
+        Assert.Null(defaultOptions.MaxPartitions);
+        Assert.Null(defaultOptions.PartitionSizeBytes);
+        Assert.False(defaultOptions.DataBoostEnabled);
+    }
+
+    [Theory, CombinatorialData]
+    public void WithDataBoostEnabled_ShouldNotAlterExistingOptionInstance(bool dataBoostEnabled)
```

Comment:
I really don't think we need this to be a combinatorial test.

I'd actually suggest changing the three earlier tests to separate out "what are the defaults" from "what do the with methods do":

For each property:
- Create a completely customized options object like in the first line of this test
- Use WithXyz to create a new object
- Assert that all values in the new object are correct

(I actually wouldn't bother checking the old ones, given that it's clearly immutable.)